### PR TITLE
chore(react-theme): refactor global shared colors

### PR DESCRIPTION
**Pull request by miroslavstastny**
## Current Behavior

In #23608 we changed `global/colors.ts` file to reduce number of shared color tokes. The problem is that this file is generated by token pipeline.
Alias shared colors are currently generated by the FUI library (not the token pipeline). For that reason the split to status, persona and other colors should be the responsibility of the FUI library as well.

## New Behavior

Reverts `global/colors.ts` to the shape generated by token pipeline, splits to status, persona and other colors in `alias/colorPalette.ts`

## Related Issue(s)

Part of #24161.

### CODE REVIEW

File: change/@fluentui-react-theme-54c0704a-5974-4449-b128-7629e3c0566b.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "none",
```
Comment (miroslavstastny) [inline change/@fluentui-react-theme-54c0704a-5974-4449-b128-7629e3c0566b.json:2]: Does not affect public API surface.

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                                                  |     Baseline (minified/GZIP) |                           PR |                                                                                                                                                                                                         Change |
| :------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------: | ---------------------------: | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Combobox.fixture.js'>Combobox (including child components)</abbr>                      | `72.723 kB`<br />`23.746 kB` | `72.762 kB`<br />`23.757 kB` |         `39 B` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/increase.png" /><br />`11 B` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/increase.png" /> |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Dropdown.fixture.js'>Dropdown (including child components)</abbr>                      | `70.792 kB`<br />`23.233 kB` | `71.026 kB`<br />`23.291 kB` |        `234 B` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/increase.png" /><br />`58 B` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/increase.png" /> |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr> | `32.895 kB`<br />`10.778 kB` | `36.266 kB`<br />`11.842 kB` | `3.371 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/increase.png" /><br />`1.064 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/increase.png" /> |
| <samp>react-theme</samp> <br /> <abbr title='bundle-size/TeamsAllThemes.fixture.js'>Teams: all themes</abbr>                                       |  `29.224 kB`<br />`6.255 kB` |  `32.555 kB`<br />`7.261 kB` | `3.331 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/increase.png" /><br />`1.006 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/increase.png" /> |
| <samp>react-theme</samp> <br /> <abbr title='bundle-size/TeamsLightTheme.fixture.js'>Teams: Light theme</abbr>                                     |   `17.088 kB`<br />`4.89 kB` |   `20.46 kB`<br />`5.884 kB` |    `3.372 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/increase.png" /><br />`994 B` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/increase.png" /> |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-accordion</samp> <br /> <abbr title='bundle-size/Accordion.fixture.js'>Accordion (including children components)</abbr>                                           |  `79.562 kB`<br />`24.121 kB` |
| <samp>react-alert</samp> <br /> <abbr title='bundle-size/Alert.fixture.js'>Alert</abbr>                                                                                       |   `83.79 kB`<br />`20.841 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/Avatar.fixture.js'>Avatar</abbr>                                                                                    |  `48.283 kB`<br />`13.644 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroup.fixture.js'>AvatarGroup</abbr>                                                                          |   `15.072 kB`<br />`6.016 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroupItem.fixture.js'>AvatarGroupItem</abbr>                                                                  |  `68.464 kB`<br />`19.067 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/Badge.fixture.js'>Badge</abbr>                                                                                       |   `22.503 kB`<br />`7.153 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/CounterBadge.fixture.js'>CounterBadge</abbr>                                                                         |   `23.406 kB`<br />`7.449 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/PresenseBadge.fixture.js'>PresenceBadge</abbr>                                                                       |   `23.947 kB`<br />`7.022 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/Button.fixture.js'>Button</abbr>                                                                                    |   `36.396 kB`<br />`9.575 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/CompoundButton.fixture.js'>CompoundButton</abbr>                                                                    |  `43.469 kB`<br />`10.812 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/MenuButton.fixture.js'>MenuButton</abbr>                                                                            |  `39.014 kB`<br />`10.456 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/SplitButton.fixture.js'>SplitButton</abbr>                                                                          |   `46.544 kB`<br />`11.84 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/ToggleButton.fixture.js'>ToggleButton</abbr>                                                                        |   `51.91 kB`<br />`11.003 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.All.fixture.js'>Card - All</abbr>                                                                                |  `67.458 kB`<br />`19.264 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.fixture.js'>Card</abbr>                                                                                          |   `63.14 kB`<br />`18.176 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardFooter.fixture.js'>CardFooter</abbr>                                                                              |    `8.461 kB`<br />`3.555 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardHeader.fixture.js'>CardHeader</abbr>                                                                              |    `9.504 kB`<br />`3.896 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardPreview.fixture.js'>CardPreview</abbr>                                                                            |     `8.562 kB`<br />`3.61 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `189.031 kB`<br />`51.967 kB` |
| <samp>react-dialog</samp> <br /> <abbr title='bundle-size/Dialog.fixture.js'>Dialog (including children components)</abbr>                                                    |  `85.574 kB`<br />`25.521 kB` |
| <samp>react-divider</samp> <br /> <abbr title='bundle-size/Divider.fixture.js'>Divider</abbr>                                                                                 |   `16.359 kB`<br />`5.853 kB` |
| <samp>react-image</samp> <br /> <abbr title='bundle-size/Image.fixture.js'>Image</abbr>                                                                                       |    `10.68 kB`<br />`4.215 kB` |
| <samp>react-input</samp> <br /> <abbr title='bundle-size/Input.fixture.js'>Input</abbr>                                                                                       |   `23.554 kB`<br />`7.644 kB` |
| <samp>react-label</samp> <br /> <abbr title='bundle-size/Label.fixture.js'>Label</abbr>                                                                                       |    `9.238 kB`<br />`3.815 kB` |
| <samp>react-link</samp> <br /> <abbr title='bundle-size/Label.fixture.js'>Link</abbr>                                                                                         |   `12.231 kB`<br />`4.925 kB` |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.fixture.js'>Menu (including children components)</abbr>                                                          | `115.929 kB`<br />`35.379 kB` |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.Selectable.fixture.js'>Menu (including selectable components)</abbr>                                             | `119.128 kB`<br />`35.874 kB` |
| <samp>react-overflow</samp> <br /> <abbr title='bundle-size/hooks.fixture.js'>hooks only</abbr>                                                                               |   `10.898 kB`<br />`4.174 kB` |
| <samp>react-popover</samp> <br /> <abbr title='bundle-size/Popover.fixture.js'>Popover</abbr>                                                                                 |  `103.05 kB`<br />`31.561 kB` |
| <samp>react-portal</samp> <br /> <abbr title='bundle-size/Portal.fixture.js'>Portal</abbr>                                                                                    |   `10.576 kB`<br />`3.875 kB` |
| <samp>react-positioning</samp> <br /> <abbr title='bundle-size/usePopper.fixture.js'>usePositioning</abbr>                                                                    |     `19.7 kB`<br />`7.404 kB` |
| <samp>react-provider</samp> <br /> <abbr title='bundle-size/FluentProvider.fixture.js'>FluentProvider</abbr>                                                                  |   `15.655 kB`<br />`5.835 kB` |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/Radio.fixture.js'>Radio</abbr>                                                                                       |      `36.238 kB`<br />`12 kB` |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/RadioGroup.fixture.js'>RadioGroup</abbr>                                                                             |   `14.361 kB`<br />`5.728 kB` |
| <samp>react-select</samp> <br /> <abbr title='bundle-size/Select.fixture.js'>Select</abbr>                                                                                    |   `20.746 kB`<br />`7.299 kB` |
| <samp>react-slider</samp> <br /> <abbr title='bundle-size/Slider.fixture.js'>Slider</abbr>                                                                                    |   `32.07 kB`<br />`10.033 kB` |
| <samp>react-spinbutton</samp> <br /> <abbr title='bundle-size/SpinButton.fixture.js'>SpinButton</abbr>                                                                        |  `43.899 kB`<br />`12.362 kB` |
| <samp>react-spinner</samp> <br /> <abbr title='bundle-size/Spinner.fixture.js'>Spinner</abbr>                                                                                 |   `19.848 kB`<br />`6.384 kB` |
| <samp>react-switch</samp> <br /> <abbr title='bundle-size/Switch.fixture.js'>Switch</abbr>                                                                                    |  `32.562 kB`<br />`10.253 kB` |
| <samp>react-text</samp> <br /> <abbr title='bundle-size/Text.Default.fixture.js'>Text - Default</abbr>                                                                        |   `11.682 kB`<br />`4.561 kB` |
| <samp>react-text</samp> <br /> <abbr title='bundle-size/Text.Wrappers.fixture.js'>Text - Wrappers</abbr>                                                                      |   `14.992 kB`<br />`4.995 kB` |
| <samp>react-textarea</samp> <br /> <abbr title='bundle-size/Textarea.fixture.js'>Textarea</abbr>                                                                              |    `23.674 kB`<br />`7.83 kB` |
| <samp>react-theme</samp> <br /> <abbr title='bundle-size/SingleThemeToken.fixture.js'>Single theme token import</abbr>                                                        |            `69 B`<br />`89 B` |
| <samp>react-tooltip</samp> <br /> <abbr title='bundle-size/Tooltip.fixture.js'>Tooltip</abbr>                                                                                 |  `41.504 kB`<br />`14.622 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/4de0899dd07982aabdc611b1837bf9acfca44efc'>4de0899dd07982aabdc611b1837bf9acfca44efc</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24435/perf-test-react-components/Avatar-mount_base.html">1425</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24435/perf-test-react-components/Avatar-mount.html">1441</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24435/perf-test-react-components/Button-mount_base.html">1107</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24435/perf-test-react-components/Button-mount.html">1112</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24435/perf-test-react-components/FluentProvider-mount_base.html">1815</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24435/perf-test-react-components/FluentProvider-mount.html">1832</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24435/perf-test-react-components/FluentProviderWithTheme-mount_base.html">711</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24435/perf-test-react-components/FluentProviderWithTheme-mount.html">713</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24435/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">685</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24435/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">667</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24435/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">730</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24435/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">735</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24435/perf-test-react-components/MakeStyles-mount_base.html">2241</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24435/perf-test-react-components/MakeStyles-mount.html">2230</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24435/perf-test-react-components/SpinButton-mount_base.html">2853</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24435/perf-test-react-components/SpinButton-mount.html">2895</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by miroslavstastny** [comment]
I do not like the bundle size increase, exploring another ways...

**Comment by miroslavstastny** [comment]
By exporting all the global shared colors individually from `packages/react-components/react-theme/src/global/colors.ts` instead of through the `sharedColors` object the bundle size is ~back at the original numbers.

We might need to refactor the approach agains once we explore the ways to make the global colors public (#23562).

**Review by layershifter — APPROVED** [review]

**Pull request merged**